main.tf

```diff
@@ -46,42 +46,58 @@ resource "google_compute_region_backend_service" "default" {
   protocol         = "${var.ip_protocol}"
   timeout_sec      = 10
   session_affinity = "${var.session_affinity}"
-  backend          = ["${var.backends}"]
+  dynamic "backend" {
+    iterator = group
+    for_each = var.backends
+    content {
+      group = group.value["group"]
+    }
+  }
   health_checks    = ["${element(compact(concat(google_compute_health_check.tcp.*.self_link,google_compute_health_check.http.*.self_link)), 0)}"]
 }
 
 resource "google_compute_health_check" "tcp" {
-  count = "${var.http_health_check ? 0 : 1}"
+  count = "${var.health_check["type"] == "tcp" ? 1 : 0}"
   project = "${var.project}"
   name    = "${var.name}-hc"
-
   tcp_health_check {
     port = "${var.health_port}"
   }
 }
 
 resource "google_compute_health_check" "http" {
-  count = "${var.http_health_check ? 1 : 0}"
+  count = "${var.health_check["type"] == "http" ? 1 : 0}"
   project = "${var.project}"
   name    = "${var.name}-hc"
-
+  check_interval_sec = "${var.health_check["check_interval_sec"]}" # - (Optional) How often (in seconds) to send a health check. The default value is 5 seconds.
```

Comment:
`check_interval`, `healthy_threshold`, `timeout_sec`, and `unhealthly_threshold`, should be configured on both `google_compute_health_check` resources.